src/dials/command_line/stills_process.py

```diff
@@ -1190,33 +1213,59 @@ def index(self, experiments, reflections):
-            )
-            idxr.index()
-        elif not indexing_succeeded:
-            indexing_error = None
-            for method in params.indexing.stills.method_list:
-                params.indexing.method = method
+        if not indexing_succeeded:
+            if self.params.indexing.stills.reflection_subsampling.enable:
+                subsets = range(
+                    self.params.indexing.stills.reflection_subsampling.step_start,
+                    self.params.indexing.stills.reflection_subsampling.step_stop
+                    - self.params.indexing.stills.reflection_subsampling.step_size,
+                    -self.params.indexing.stills.reflection_subsampling.step_size,
+                )
+            else:
+                subsets = [100]
+            all_reflections = reflections
+            subset_indexing_error = None
+            for i in subsets:
+                if i != 100:
+                    reflections = all_reflections.select(
+                        flex.random_permutation(len(all_reflections))
+                    )[: int(len(all_reflections) * i / 100)]
                 try:
-                    idxr = Indexer.from_parameters(
-                        reflections, experiments, params=params
-                    )
-                    idxr.index()
-                except Exception as e:
-                    logger.info("Couldn't index using method %s", method)
-                    if indexing_error is None:
-                        if e is None:
-                            e = Exception(f"Couldn't index using method {method}")
-                        indexing_error = e
+                    if params.indexing.stills.method_list is None:
+                        idxr = Indexer.from_parameters(
+                            reflections,
+                            experiments,
+                            params=params,
+                        )
+                        idxr.index()
+                    else:
+                        ml_indexing_error = None
+                        for method in params.indexing.stills.method_list:
+                            params.indexing.method = method
+                            try:
+                                idxr = Indexer.from_parameters(
+                                    reflections,
+                                    experiments,
+                                    known_crystal_models=known_crystal_models,
+                                    params=params,
+                                )
+                                idxr.index()
+                            except Exception as e_ml:
+                                logger.info("Couldn't index using method %s", method)
+                                ml_indexing_error = e_ml
+                            else:
+                                ml_indexing_error = None
+                                break
+                        if ml_indexing_error is not None:
```

Comment:
For consistency with line 1267 below

```suggestion
                        if ml_indexing_error:
```